### ISSUE TITLE
What is the difference between update() and write()?

### ISSUE THREAD

**Issue by kh75park** [Zazama/node-id3#70]
I have a data like this that I am using to call update() and write().

```
const newMp3Tags: any = {
	title: "myTitle",
	artist: "myArtist",
	album: "myAlbum"
};
```

Calling update() and write() seems to have exact same behavior.

I would have thought that write() will remove all other tag data and just keep my data as tag data.
And update() would just update the tags that I have in my data and keep all others as is.
But whether or not I call update() or write(), I see all other data as if update() was called.

**Comment by Zazama** [comment]
Hello,

you're describing both functions correctly,
if I have an mp3 file with the following tags already in it:

```
{
  title: "a",
  artist: "b",
  album: "c",
  year: 2020
}
```

using `NodeID3.write(newMp3Tags, file)` will produce

```
  title: "myTitle",
  artist: "myArtist",
  album: "myAlbum"
```

but using `NodeID3.update(newMp3Tags, file)` will produce

```
  title: "myTitle",
  artist: "myArtist",
  album: "myAlbum",
  year: 2020
```

Locally, everything works as expected for me, but if you have problems, you can send me an mp3 and, importantly, your code calling NodeID3 via GitHub or email to jan.metzger@gmx.net

**Comment by kh75park** [comment]
Hi.
OK. You are right. Data-wise, it works as expected.
Maybe it's the WinAmp that's doing something strange.
I set the MP3 file with track and year along with some other basic tag data to start.
Then I wrote using new tag data without track and year.  Your API gives correct result:

> tags.ori={"raw":{"TYER":"2020","TALB":"OAlb","TPE1":"OArt","TIT2":"OT","TRCK":"01"},"year":"2020","album":"OAlb","artist":"OArt","title":"OT","trackNumber":"01"}
> testing write()
> tagsAfterWrite={"raw":{"TIT2":"newTitle","TPE1":"newArtist","TALB":"newAlbum"},"title":"newTitle","artist":"newArtist","album":"newAlbum"}
> 

But strangely, when I open the MP3 file using WinAmp and view the file info, I still see the track and year.  One thing though I just noticed is that in "Basic Info" tab, I see my new data + track and year.  In "ID3v1" tab, I see the original data + track and year.  And in "ID3v2" tab, I see the new data only without the track and year.

Based on this, I think  your code seems to work as designed and WinAmp may be doing something strange.  Thank you.
